aws/resource_aws_dynamodb_table_item_test.go

```diff
@@ -0,0 +1,363 @@
+					resource.TestCheckResourceAttr("aws_dynamodb_table_item.test", "item", itemAfter+"\n"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAWSDynamoDbItemDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).dynamodbconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_dynamodb_table_item" {
+			continue
+		}
+
+		attrs := rs.Primary.Attributes
+		attributes, err := expandDynamoDbTableItemAttributes(attrs["item"])
+		if err != nil {
+			return err
+		}
+
+		result, err := conn.GetItem(&dynamodb.GetItemInput{
+			TableName:                aws.String(attrs["table_name"]),
+			ConsistentRead:           aws.Bool(true),
+			Key:                      buildDynamoDbTableItemQueryKey(attributes, attrs["hash_key"], attrs["range_key"]),
+			ProjectionExpression:     buildDynamoDbProjectionExpression(attributes),
+			ExpressionAttributeNames: buildDynamoDbExpressionAttributeNames(attributes),
+		})
+		if err != nil {
+			if isAWSErr(err, dynamodb.ErrCodeResourceNotFoundException, "") {
+				return nil
+			}
+			return fmt.Errorf("Error retrieving DynamoDB table item: %s", err)
+		}
+		if result.Item == nil {
+			return nil
+		}
+
+		return fmt.Errorf("DynamoDB table item %s still exists.", rs.Primary.ID)
+	}
+
+	return nil
+}
+
+func testAccCheckAWSDynamoDbTableItemExists(n string, item *dynamodb.GetItemOutput) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No DynamoDB table item ID specified!")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).dynamodbconn
+
+		attrs := rs.Primary.Attributes
+		attributes, err := expandDynamoDbTableItemAttributes(attrs["item"])
+		if err != nil {
+			return err
+		}
+
+		result, err := conn.GetItem(&dynamodb.GetItemInput{
+			TableName:                aws.String(attrs["table_name"]),
+			ConsistentRead:           aws.Bool(true),
+			Key:                      buildDynamoDbTableItemQueryKey(attributes, attrs["hash_key"], attrs["range_key"]),
+			ProjectionExpression:     buildDynamoDbProjectionExpression(attributes),
+			ExpressionAttributeNames: buildDynamoDbExpressionAttributeNames(attributes),
+		})
+		if err != nil {
+			return fmt.Errorf("[ERROR] Problem getting table item '%s': %s", rs.Primary.ID, err)
+		}
+
+		*item = *result
+
+		return nil
+	}
+}
+
+func testAccCheckAWSDynamoDbTableItemCount(tableName string, count int) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := testAccProvider.Meta().(*AWSClient).dynamodbconn
+		out, err := conn.Scan(&dynamodb.ScanInput{
+			ConsistentRead: aws.Bool(true),
+			TableName:      aws.String(tableName),
+			Select:         aws.String(dynamodb.SelectCount),
+		})
+		if err != nil {
+			return err
+		}
+		expectedCount := int64(count)
```

Comment:
Nitpick: I think we can rid of this line if we just `int64` the type in the signature. 😄